frontend/packages/console-app/src/components/nodes/NodesPage.tsx

```diff
@@ -132,27 +264,106 @@ const NodesTableRow = connect<NodesRowMapFromStateProps, null, NodesTableRowProp
     return (
       <TableRow id={nodeUID} index={index} trKey={rowKey} style={style}>
-        <TableData className={tableColumnClasses[0]}>
+        <TableData className={nodeColumnInfo.name.classes}>
           <ResourceLink kind={referenceForModel(NodeModel)} name={nodeName} title={nodeUID} />
         </TableData>
-        <TableData className={tableColumnClasses[1]}>
+        <TableData
+          className={nodeColumnInfo.status.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.status.id}
+        >
           <NodeStatus node={node} showPopovers />
         </TableData>
-        <TableData className={tableColumnClasses[2]}>
+        <TableData
+          className={nodeColumnInfo.role.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.role.id}
+        >
           <NodeRoles node={node} />
         </TableData>
-        <TableData className={tableColumnClasses[3]}>{pods}</TableData>
-        <TableData className={tableColumnClasses[4]}>{memory}</TableData>
-        <TableData className={tableColumnClasses[5]}>
+        <TableData
+          className={nodeColumnInfo.pods.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.pods.id}
+        >
+          {pods}
+        </TableData>
+        <TableData
+          className={nodeColumnInfo.memory.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.memory.id}
+        >
+          {memory}
+        </TableData>
+        <TableData
+          className={nodeColumnInfo.cpu.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.cpu.id}
+        >
           {cores ? `${formatCores(cores)} cores` : '-'}
         </TableData>
-        <TableData className={tableColumnClasses[6]}>{storage}</TableData>
-        <TableData className={tableColumnClasses[7]}>
+        <TableData
+          className={nodeColumnInfo.filesystem.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.filesystem.id}
+        >
+          {storage}
+        </TableData>
+        <TableData
+          className={nodeColumnInfo.created.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.created.id}
+        >
           <Timestamp timestamp={node.metadata.creationTimestamp} />
         </TableData>
-        <TableData className={tableColumnClasses[8]}>
+        <TableData
+          className={nodeColumnInfo.instancetype.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.instancetype.id}
+        >
+          {instanceType}
+        </TableData>
+        <TableData
+          className={nodeColumnInfo.machine.classes}
+          columns={columns}
+          columnID={nodeColumnInfo.machine.id}
+        >
+          <ResourceLink
+            kind={referenceForModel(MachineModel)}
+            name={machine.name}
+            namespace={machine.namespace}
+          />
```

Comment:
You want to display `-` if there is no machine. (It won't always be set.)

Reply:
Done.